### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: '/home/MHC_Class_I/8/pvf_sample_98_arr.all_epitopes.tsv

### ISSUE THREAD

**Issue by min-codes** [griffithlab/pVACtools#1004]
### Installation Type

Docker

### pVACtools Version / Docker Image

griffithlab/pvactools:3.1.2

### Describe the bug

Hi there, I would like any suggestion on this error message I encountered. 

(1) I used parallel command to run pVacFuse using the pvactools docker image to predict HLA-I binding abilities of fusion transcripts detected by Arriba and FusionCatcher, for a large number of samples. I included some HLA-I alleles, and selected 5 prediction tools (NetMHCpan MHCflurry MHCnuggetsI NNalign PickPocket) and range of k-mer length is set to 8 to 14. Most of the samples worked fine, some however returned this error message of "No output files were created. Aborting." right after the "Parsing binding predictions for HLAxx". Please see below the last few lines of the output message from pVacFuse. Since other samples worked, I assume the problem is not with the code I used. Could it be that the RAM was maxed out at the time?

(2) Another quick question - how do we customise sample-name using pVacFuse option -a ? The tool does not seem to like any other string value apart from "sample_name" 

Thanks in advance,
Min

### How to reproduce this bug

```Shell
docker run --name ${num}_${tool}_${freq}_${ft}_sample_${ID} -v ${workdir}/NativeFilesForNeoantigenPrediction/${subdir}/${num}_${tool}_${freq}_${ft}:/home/min/ griffithlab/pvactools:3.1.2 pvacfuse run \
-a sample_name -e1 8,9,10,11,12,13,14 --iedb-install-directory /opt/iedb -k -t 16 --net-chop-method cterm --netmhc-stab --run-reference-proteome-similarity /home/min/agf_sample_${ID}_${tool} pvf_sample_${ID}_${tool} HLA-A*11:01,HLA-A*24:02,HLA-A*33:03,HLA-A*02:07,HLA-A*11:353,HLA-A*02:01,HLA-A*02:03,HLA-A*11:02,HLA-A*11:12,HLA-B*40:01,HLA-B*46:01,HLA-B*58:01,HLA-B*13:01,HLA-B*15:02,HLA-B*38:02,HLA-B*51:01,HLA-B*54:01,HLA-B*27:04,HLA-B*15:01,HLA-C*07:02,HLA-C*01:02,HLA-C*03:04,HLA-C*08:01,HLA-C*03:02,HLA-C*15:02,HLA-C*12:02,HLA-C*04:01,HLA-C*14:02,HLA-C*06:02,HLA-C*04:03 NetMHCpan MHCflurry MHCnuggetsI NNalign PickPocket /home/min/pvf_sample_${ID}_${tool}
```

### Input files

(input into AGFusion)
Native output from (fusions.tsv) from Arriba

(stdout from AGFusion)
2023-07-13 16:30:48,753 - AGFusion - WARNING - Output directory /6_arr_16_ROPN1B--ALG1L/agf_sample_98_arr already exists! Overwriting...
WARNING:AGFusion:Output directory /6_arr_16_ROPN1B--ALG1L/agf_sample_98_arr already exists! Overwriting...
2023-07-13 16:30:48,762 - AGFusion - INFO - Read 1 fusions from the file.
INFO:AGFusion:Read 1 fusions from the file.
INFO:pyensembl.sequence_data:Loaded sequence dictionary from /home/minhui/.cache/pyensembl/GRCh38/ensembl95/Homo_sapiens.GRCh38.cdna.all.fa.gz.pickle
INFO:pyensembl.sequence_data:Loaded sequence dictionary from /home/minhui/.cache/pyensembl/GRCh38/ensembl95/Homo_sapiens.GRCh38.ncrna.fa.gz.pickle

(content of directory generated by AGFusion)
ENST00000251776_ENST00000340333.exon.png
ENST00000504401_ENST00000340333.png
ENST00000251776_ENST00000340333.png
ENST00000514116_ENST00000340333.exon.png
ENST00000504401_ENST00000340333.exon.png
ENST00000514116_ENST00000340333.png
ROPN1B_ALG1L.exons.csv
ROPN1B_ALG1L_cdna.fa
ROPN1B_ALG1L.fusion_transcripts.csv
ROPN1B_ALG1L_cds.fa
ROPN1B_ALG1L.protein_domains.csv
ROPN1B_ALG1L_protein.fa

### Log output

```
Executing MHC Class I predictions
Converting Fusion file to TSV
Completed
Generating Variant Peptide FASTA and Key File
Completed
Parsing the Variant Peptide FASTA and Key File
Completed
Calculating Manufacturability Metrics
Completed
Splitting FASTA into smaller chunks
Splitting FASTA into smaller chunks - Entries 1-3
Completed
Fasta file /home/MY_STUFF/pvf_sample_98_arr/MHC_Class_I/8/tmp/pvf_sample_98_arr.8.fa.split_1-3 is empty. Skipping
Parsing binding predictions for Allele HLA-A*02:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*02:03 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*02:07 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*11:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*11:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*11:12 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*11:353 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*24:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-A*33:03 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*13:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*15:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*15:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*27:04 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*38:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*40:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*46:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*51:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*54:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-B*58:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*01:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*03:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*03:04 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*04:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*04:03 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*06:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*07:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*08:01 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*12:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*14:02 and Epitope Length 8 - Entries 1-3
Parsing binding predictions for Allele HLA-C*15:02 and Epitope Length 8 - Entries 1-3
No output files were created. Aborting.
Traceback (most recent call last):
  File "/usr/local/bin/pvacfuse", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/pvactools/tools/pvacfuse/main.py", line 94, in main
    args[0].func.main(args[1])
  File "/usr/local/lib/python3.7/site-packages/pvactools/tools/pvacfuse/run.py", line 216, in main
    append_columns(intermediate_output_file, "{}.tsv".format(input_file), output_file)
  File "/usr/local/lib/python3.7/site-packages/pvactools/tools/pvacfuse/run.py", line 93, in append_columns
    with open(intermediate_output_file, 'r') as input_fh, open(output_file, 'w') as output_fh:
FileNotFoundError: [Errno 2] No such file or directory: '/home/MY_STUFF/pvf_sample_98_arr/MHC_Class_I/8/pvf_sample_98_arr.all_epitopes.tsv'
```

### Output files

(Content of the output directory - only one directory generated, and for k-mer length 8)
log/
tmp/
pvf_sample_98_arr.fa
pvf_sample_98_arr.fa.tsv

**Comment by susannasiebert** [comment]
Hi @min-codes,

Thank you for your interest in pVACtools. I apologize that you're running into an error with our pVAC fuse software.

(1) Would you be able to attach your actual AGFusion output directory to this ticket so I can try to reproduce this issue on my end?

(2) The `-a` option takes the literal string "sample_name" to indicate that you want to output the sample name as an additional report column; not what the sample name itself should be. When this is set the sample name from your positional sample name parameter is written in the sample name output column (in your case `pvf_sample_98_arr`).

**Comment by min-codes** [comment]
Hi @susannasiebert , thank you for your reply. GitHub doesn't let me upload the fasta files, so I uploaded them to my [gDrive ](https://drive.google.com/drive/folders/1PaEKOvqVLA6D0jSTvuaEkeBAlyO3wsaK?usp=sharing) and attached the link here. These files are for another sample, as I accidentally removed the ones I posted here when troubleshooting. (this sample also had the same error message from pVacFuse)

gDrive folder content:
1. **sample_103_fc.tsv** : Output file from FusionCatcher, input for AGFusion.)
2. **sample_103_fc.rerun.log.txt** : stdout from running AGFusion and pVacFuse
3. **agf_sample_103_fc** : output directory of AGFusion
4. **pvf_sample_103_fc** : output directory of pVacFuse

FYI, I used AGFusion v1.4.1 with database v95 / Ensembl v95 as that was the latest available for AGFusion. Python v3.8.10. This is the code i used to run AGFusion

```
agfusion batch \
-f sample_103_fc.tsv \
-a fusioncatcher \
-db agfusion.homo_sapiens.95.db \
-o agf_sample_103_fc \
--middlestar \
--noncanonical
```

**Comment by susannasiebert** [comment]
Thank you for providing your sample data. I'm at a conference this week but I will debug this on my end when I'm back at work next week.

RE (2), that is correct. You literally say `-a sample_name` (where sample_is does _not_ get replaced with your sample name). This tells pVACfuse to take the second positional parameter from your pVACfuse command and use that to fill in the sample name column. If you want to use "test_sample_103_fc" as your sample name, you will need to use it in place of the second positional parameter, e.g. `pvacfuse run /home/min/agf_sample_${ID}_${tool} test_sample_103_fc [..] -a sample_name`

**Comment by susannasiebert** [comment]
It looks like all the fusions in your AGfusion input (specifically `MOG-29672339_DDOST-20651804/MOG_DDOST_protein.fa`) are missing the protein sequence to the right of the fusion (the fusion break point `*` is at the end of the protein sequence). These are then skipped in pVACfuse. The cds file shows the 5' nucleotides as TGA, which is a stop codon, so that is probably why.

I'm working on a better error message in pVACfuse.

**Comment by susannasiebert** [comment]
This error should be fixed in version 4.0.2 and your pVACfuse run should complete, although without any results since there are no processable fusions. I'm closing this issue but please feel free to reopen it, should you still run into problems.
